src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxrResolver.cpp

```diff
@@ -89,88 +89,81 @@ HostFxrResolver::GetHostFxrParameters(
             size_t size = MAX_PATH * 2;
             hostfxrPath.resize(size);
 
-            get_hostfxr_parameters params;
-            get_hostfxr_parameters* pParams = NULL;
-
             int result = -1;
 
             // when dotnet launched from path case, we don't want to set the dotnet_root
             if (!equals_ignore_case(expandedProcessPath, L"dotnet.exe")) {
+                get_hostfxr_parameters params;
                 params.size = sizeof(get_hostfxr_parameters);
                 params.assembly_path = applicationPhysicalPath.c_str();
                 params.dotnet_root = expandedProcessPath.parent_path().c_str();
```

Comment:
I believe the return of `parent_path()` is going to be a [temporary object](https://en.cppreference.com/w/cpp/language/lifetime#Temporary_object_lifetime) here, so it will still be destructed before the `get_hostfxr_path` call (resulting in garbage in `params.dotnet_root`).

Reply:
Okay going to show my C++ noob here, what's the proper pattern to do this call then, do I need to explicitly copy the string into another variable, like hostFxrPath for it not to get destroyed?

Reply:
That is an option. I would just assign the return of `parent_path()`  to a variable such that you keep it around until you call `get_hostfxr_path`. For example:
```c++
{
    fs::path process_folder = expandedProcessPath.parent_path();
    params.dotnet_root = process_folder.c_str();
    result = get_hostfxr_path(...)
} // process_folder goes out of scope and is destructed - we don't need it anymore after get_hostfxr_path is done
```

Reply:
If you are interested in seeing exactly what is going on, one of the things I usually do is just debug through a simple case in a console app. That would give you an easy way step though the difference between two cases like:
```c++
std::filesystem::path my_path = ...
{
    std::filesystem::path parent = my_path.parent_path();
    const wchar_t* ptr = parent.c_str();
    std::wcout << ptr << std::endl;
} // parent goes out of scope and is destructed
{
    const wchar_t* ptr = my_path.parent_path().c_str();
    // object returned by parent_path is immediately destructed after the call in the line above
    std::wcout << ptr << std::endl;
}
```
You can do something like look at the [Disassembly Window](https://docs.microsoft.com/visualstudio/debugger/how-to-use-the-disassembly-window) or set a [Function Breakpoint](https://docs.microsoft.com/visualstudio/debugger/using-breakpoints#BKMK_Set_a_breakpoint_in_a_source_file) for the destructor (`std::filesystem::path::~path`) to see when the path object is being destructed.

Reply:
Thanks for the pointer, hopefully that this care of the non deterministic failures finally!